test/test_local_io.py

```diff
@@ -280,29 +292,28 @@ def is_nonempty_json(path_and_stream):
             len(json_dp)
 
     def test_saver_iterdatapipe(self):
-        def filepath_fn(name: str) -> str:
-            return os.path.join(self.temp_dir.name, os.path.basename(name))
-
         # Functional Test: Saving some data
         name_to_data = {"1.txt": b"DATA1", "2.txt": b"DATA2", "3.txt": b"DATA3"}
         source_dp = IterableWrapper(sorted(name_to_data.items()))
         saver_dp = source_dp.save_to_disk(filepath_fn=filepath_fn, mode="wb")
```

Comment:
Original `filepath_fn` only takes a single parameter, but now it requires two. So, we have to use `partial` from python `functools` to add `self.temp_dir.name` to this new function.
```suggestion
        saver_dp = source_dp.save_to_disk(filepath_fn=partial(filepath_fn, self.temp_dir.name), mode="wb")`
```